streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3430,24 +3430,67 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
 
   /**
    * Submerges the error case of an `Either` into the `ZStream`.
+   *
+   * {{{
+   * ZStream.absolve(ZStream(Right(1), Right(2), Right(3)))
+   * // 1, 2, 3
+   *
+   * ZStream
+   *  .absolve(ZStream(Right(1), Left("error"), Right(3)))
+   *  .catchSome { case s: String => ZStream(s) }
+   * // 1, error
+   * }}}
    */
   def absolve[R, E, O](xs: ZStream[R, E, Either[E, O]]): ZStream[R, E, O] =
     xs.mapM(ZIO.fromEither(_))
 
   /**
    * Accesses the environment of the stream.
+   *
+   * {{{
+   * ZStream
+   *   .access[Has[String]](_.get)
+   *   .provideSomeLayer[ZEnv](ZLayer.succeed("str"))
+   * // str
+   * }}}
    */
   def access[R]: AccessPartiallyApplied[R] =
     new AccessPartiallyApplied[R]
 
   /**
    * Accesses the environment of the stream in the context of an effect.
+   *
+   * {{{
+   * trait A {
+   *  def a: UIO[String]
+   * }
+   *
+   * ZStream
+   *  .accessM[Has[A]](_.get.a)
```

Comment:
Consider `.serviceWith[A](_.a)` for brevity, here and elsewhere.

Reply:
How about this?

```scala
ZStream
  .accessM[A](_.a)
  .provideSome[ZEnv](_ =>
    new A {
      override def a: UIO[String] = UIO("str")
    }
  )
```

I want to use `accessM` here to describe that method.